presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -148,7 +148,7 @@ public static PlanNodeStatsEstimate intersect(PlanNodeStatsEstimate left, PlanNo
                             .setNullsFraction(min(leftSymbolStats.getNullsFraction(), rightSymbolStats.getNullsFraction()))
                             .build());
         }
-
+        statsBuilder.setOutputRowCount(left.getOutputRowCount() + right.getOutputRowCount());  // this is the maximum row count;
```

Comment:
Since these are intermediate stats internal to this computation, it only became a problem when we started using the rowCount in groupBy.  So, I can keep it as a separate commit, but IMO it's directly related to the rest of the changes (e.g. it can't be tested independently).